src/goto-symex/symex_function.cpp

```diff
@@ -412,7 +413,7 @@ void goto_symext::symex_function_call_deref(const expr2tc &expr)
     // and then call func(NULL). Our C++ frontend detects this a failure during the parsing
     // C++ does not allow declarations without prototypes
     // TODO: see GitHub issues #584 and #585
-    if(ct.arguments.size() && ct.arguments.size() != call.operands.size())
+    if(0 && ct.arguments.size() && ct.arguments.size() != call.operands.size())
```

Comment:
`0` forces the guard to be false, can you remove it?

Shouldn't it be `ct.arguments.size() > call.operands.size()` like the previous one?

Reply:
I'm feeling uneasy about this `illegal_filter`: why is anything illegal at this point in symex? If not a valid call target, shouldn't it be guarded by some predicate? For instance, if all targets were "illegal" that would mean execution cannot continue - this should be VERIFICATION FAILED, no?

Reply:
For now we are still filtering, but I'm checking whether this filter could actually be a no-op and if not why not.